src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -441,6 +441,79 @@ public function get($id, $invalidBehavior = ContainerInterface::EXCEPTION_ON_INV
     }
 
     /**
+     * {@inheritdoc}
+     *
+     * @param bool $reload whether or not reload the type map
+     */
+    public function getServiceIdsOfType($class, $reload = true)
+    {
+        if ($reload) {
+            $this->populateAvailableTypes();
+        }
+
+        return parent::getServiceIdsOfType($class);
+    }
+
+    /**
+     * Populates the list of available types.
+     *
+     * @return array the type map
+     */
+    public function populateAvailableTypes()
```

Comment:
That's not exactly true, this method has to be called the class of a definition is updated.
What do you think of the last commit using ``\SplObserver`` and ``\SplSubject`` to remove this need ?